File: PG/[PG]다음큰숫자.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int returnCnt(int n){
    int cnt = 0;
    while(n != 0){
        if(n%2 == 1) cnt++; 
        n = n/2;
    }
    return cnt;
}

int solution(int n) {
    int cnt = returnCnt(n);
    while(true){
        if(cnt == returnCnt(++n)) return n;
    }
}

File: PG/[PG]다음큰숫자.java
import java.util.*;

public class PG_다음큰숫자 {
    public static int convertToBinary(int n){
        int cnt = 0;
        while(n != 1 && n != 0){
            int i = n % 2;
            if(i == 1) cnt++;
            n/=2;
        }
        if(n == 1) cnt++;
        return cnt;
    }

    public int solution(int n){
        int cnt = convertToBinary(n);
        int cur = n+1;
        while(convertToBinary(cur) != cnt) cur++;
        return cur;
    }

    public static void main(String[] args) {
        PG_다음큰숫자 ps = new PG_다음큰숫자();
        System.out.println(ps.solution(78));
        System.out.println(ps.solution(15));
    }
}
